src/Npgsql/NpgsqlConnector.cs

```diff
@@ -486,6 +489,18 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
                             Log.Error("Exception bubbled out of multiplexing read loop", t.Exception!, Id);
                         }, TaskContinuationOptions.OnlyOnFaulted);
                 }
+
+                if (_isKeepAliveEnabled)
+                {
+                    // Start the keep alive mechanism to work by scheduling the timer.
+                    // Otherwise, it doesn't work for cases when no query executed during
+                    // the connection lifetime in case of a new connector.
+                    lock (this)
+                    {
+                        var keepAlive = Settings.KeepAlive * 1000;
```

Comment:
nit: we can refactor this out to a StartKeepAlive method

Reply:
Don't think it's worth so...